### PR TITLE
[WIP] adds AudioMAE model

### LINKED ISSUE #27453: Audio-MAE - ViTMAE for audio
### Model description

This model is is a Self-supervised Vision Transformer that uses patch reconstruction as the spectrogram task. It extends MAE (which is already on HuggingFace) for audio. This model would be a valuable addition as there doesn't seem to be a self-supervised ViT model on HugginFace currently. AST is the closest and uses supervised pre-training. Conceptually, Audio-MAE is also simpler but achieves comparable performance in their paper.

Some differences compared to the standard MAE Model
- During pre-training local and hybrid attention mechanisms can be used.
- During fine-tuning masking is also used which differs to MAE.

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

**Implementation**
https://github.com/facebookresearch/AudioMAE
Created by Po-Yao Huang
@berniebear on Github

**Pre-trained Weights**
Available in the github repo

**Paper (Masked Autoencoders that Listen)**
https://arxiv.org/abs/2207.06405

### PR DESCRIPTION
**Pull request by ps4vs**
Fixes https://github.com/huggingface/transformers/issues/27453 Audio-MAE - ViTMAE for audio.

@ArthurZucker @sanchit-gandhi @ylacombe @pennychong94

There are some nit changes pending. Looking forward to push within 2-3 days.

Thanks @NielsRogge for awesome tutorials.

### CODE REVIEW

**Comment by ps4vs** [comment]
Hi @NielsRogge, @sanchit-gandhi, dropout layers are being non-deterministic, without which I'm getting same outputs on my implementation and official codebase. 

Any suggestions on how to proceed further?

**Comment by nileshkokane01** [comment]
> use transformers.utils.set_seed if the old and new implementations are in the same framework.
[Hugging Face doc](https://huggingface.co/docs/transformers/en/add_new_model) says so.

**Comment by NielsRogge** [comment]
Hi @ps4vs in PyTorch, you can just call `.eval()` on your model to turn off dropout.
